pkg/kubelet/kubelet.go

```diff
@@ -479,7 +480,19 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 
 	oomWatcher, err := oomwatcher.NewWatcher(kubeDeps.Recorder)
 	if err != nil {
-		return nil, err
+		if utilfeature.DefaultFeatureGate.Enabled(features.KubeletInUserNamespace) {
+			if libcontainersystem.RunningInUserNS() {
+				// oomwatcher.NewWatcher returns "open /dev/kmsg: operation not permitted" error,
+				// when running in a user namespace with sysctl value `kernel.dmesg_restrict=1`.
+				klog.V(2).InfoS("Failed to create an oomWatcher (running in UserNS, ignoring)", "err", err)
+				oomWatcher = nil
+			} else {
+				klog.ErrorS(err, "Failed to create an oomWatcher (running in UserNS, Hint: enable KubeletInUserNamespace feature flag to ignore the error)")
```

Comment:
hint is misleading as the error is printed outside the check `libcontainersystem.RunningInUserNS()`.

Reply:
updated